### ISSUE TITLE
Check uniqueness of matches

### ISSUE THREAD

**Issue by d-Rickyy-b** [d-Rickyy-b/pastepwn#180]
Currently the list of matches contains a lot of duplicates. Maybe it would be nice to remove duplicates to only have unique values

**Comment by d-Rickyy-b** [comment]
On the other hand, maybe someone would like to know how many matches of a certain word were contained in a match. Hence passing a unique list would mean passing a false result

**Comment by d-Rickyy-b** [comment]
Maybe we can use a parameter "unique_matches" which determines of the results should be unique or not.

**Comment by salmannotkhan** [comment]
how can i help please guide me regarding testing and module structure

**Comment by d-Rickyy-b** [comment]
Hi @salmannotkhan, testing can be done all locally - no pastebin access required.

Currently each analyzer in https://github.com/d-Rickyy-b/pastepwn/tree/master/pastepwn/analyzers will generate an array called "matches", that contains all the matched data from a single paste.

The corresponding action to that analyzer will receive the "matches" array.

Now I personally think it would be best to implement another method called "def unique(matches)" or something like that. It can be implemented in the BasicAnalyzer class and receives a list of matches and returns a list of matches without duplicates.
In each other analyzer we would now need to add a boolean parameter "unique_matches" or similar. If set to True it will be used to filter out duplicate values.

I hope I could give you a little bit more insights. Tell me if you need more information. Thanks for you interest!

**Comment by salmannotkhan** [comment]
Type of matches is list?

**Comment by d-Rickyy-b** [comment]
Exactly!

**Comment by salmannotkhan** [comment]
In Readme it is mentioned that to use this tool I need pastebin premium account?
There are so many analyzers in that folder can you please explain how to exactlly run this thing

**Comment by d-Rickyy-b** [comment]
Yes, you need a pastebin premium account for actually using the tool. But you don't need that for testing the code.

Each analyzer takes a [paste object](https://github.com/d-Rickyy-b/pastepwn/blob/master/pastepwn/core/paste.py) as argument to the match function (check [this](https://github.com/d-Rickyy-b/pastepwn/blob/f91a0ff64821b25ce6a1dc571bc0bb1375b97bf9/pastepwn/analyzers/regexanalyzer.py#L20-L28) match function of the generic regexAnalyzer).

Now you can implement the code necessary to close this issue in the [basicAnalyzer](https://github.com/d-Rickyy-b/pastepwn/blob/f91a0ff64821b25ce6a1dc571bc0bb1375b97bf9/pastepwn/analyzers/basicanalyzer.py) and use unit tests ([here](https://github.com/d-Rickyy-b/pastepwn/blob/f91a0ff64821b25ce6a1dc571bc0bb1375b97bf9/pastepwn/analyzers/tests/basicanalyzer_test.py)) to test the uniqueness of matches.


```python
(...)

class TestBasicAnalyzer(unittest.TestCase):
    def setUp(self):
        self.analyzer = BasicAnalyzer(None)
        self.mock_action = mock.MagicMock(spec=BasicAction)

    def test_unique():
        # your code to test functionality here

(...)
```

You can read more about unittests here: https://docs.python.org/3/library/unittest.html

To execute unittests you can use:
`python -m unittest -v <path_to_test.py>`

If there are still unanswered questions, feel free to contact me again.

**Comment by salmannotkhan** [comment]
how can i see matches list? that does it contain duplicate or not. I ran the test on basicanalyzer it just return 'OK' as output i can't see any list of matches to check where it is getting duplicates

**Comment by d-Rickyy-b** [comment]
You are right. The basic analyzer is a base class and won't return any matches. In general the match function of any other analyzer will return a list of strings of all matches

It could look like this:

["match", "password", "myemail@example.com", "match"]

You can check the other analyzer unit tests to get actually results.

**Comment by salmannotkhan** [comment]
we can use matches = set(list(matches))

**Comment by d-Rickyy-b** [comment]
That's a good idea!

**Comment by salmannotkhan** [comment]
well i tried other tests also still can't see any list

**Comment by d-Rickyy-b** [comment]
Check the [regex analyzer](https://github.com/d-Rickyy-b/pastepwn/blob/f91a0ff64821b25ce6a1dc571bc0bb1375b97bf9/pastepwn/analyzers/tests/regexanalyzer_test.py)

There you can see that I am using the result from the match() function to test if the analyzer works correctly

**Comment by salmannotkhan** [comment]
> That's a good idea!

So, we can make a function

```python
def unique(matches):
return list(set(matches))
```

**Comment by Vrim** [comment]
I believe my PR #196 closes this issue.
I noticed the changelog notification from the CI. My bad. 
Not sure if I should add or let someone else do it.

**Comment by king-phyte** [comment]
Please close this issue seeing as it is resolved

**Comment by d-Rickyy-b** [comment]
Fixed in #198
